counter-sink/src/main/java/org/springframework/cloud/stream/module/metrics/CounterSinkOptions.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.metrics;
+
+import javax.validation.constraints.AssertTrue;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+
+/**
+ * Holds configuration options for the Counter Sink.
+ *
+ * @author Eric Bottard
+ */
+@ConfigurationProperties
+public class CounterSinkOptions {
```

Comment:
I really think we should stick with "options", both because this is what we had in XD, and because "properties" feels very "static state describing" (me being French is one of my "properties") whereas "options" carries the idea that you influence behavior (eric.turnOff = true is an "option").

As a last resort, I'd settle for `CounterSinkConfigurationProperties`